spyder/plugins/application/widgets/status.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Status widget for Spyder updates.
+"""
+
+# Standard library imports
+import logging
+
+# Third party imports
+from qtpy.QtCore import Slot
+
+# Local imports
+from spyder import get_versions
+from spyder.api.widgets.status import StatusBarWidget
+from spyder.config.base import _, is_pynsist
+from spyder.plugins.application.widgets.install import (
+    UpdateInstallerDialog, NO_STATUS, DOWNLOADING_INSTALLER, INSTALLING,
+    FINISHED, PENDING, CHECKING, CANCELLED)
+from spyder.utils.icon_manager import ima
+
+logger = logging.getLogger(__name__)
+
+
+class ApplicationUpdateStatus(StatusBarWidget):
+    """Status bar widget for Application update status."""
+    BASE_TOOLTIP = _("Application update status")
+    ID = 'application_update_status'
+
```

Comment:
Following another suggestion from @CAM-Gerlach lets add the possibility to call the `Check from updates...` action  from the status bar. For that we could add here a signal that then we can connect from the container to the `check_update` function:

```suggestion

    sig_check_for_updates_requested = Signal()

```